frontend/src/components/ViewActivities/activity.js

```diff
@@ -1,25 +1,106 @@
-import Card from 'react-bootstrap/Card';
 import * as time from '../Utils/time.js';
 import * as DB from '../../constants/database.js'
 import '../../styles/activities.css';
-import Accordion from 'react-bootstrap/Accordion';
+import { getField } from './activityfns.js';
+import { Accordion, Button, Card, Col, Form, Row } from 'react-bootstrap';
 
+/**
+ * Returns a dropdown of all the timezones.
+ */
+function timezonePicker() {
+  // TODO: Make this dropdown. (#51)
+  return <div></div>
+}
+
+function submitEdit(k) {
+  console.log(k);
+}
+
+/**
+ * A single activity. 
+ * 
+ * @param {Object} props This component expects the following props:
+ * - `activity` {Object} The activity to display.
+ */
 class Activity extends React.Component {
+  /** {@inheritdoc} */
+  constructor(props) {
+    super(props);
+
+    this.state = { editing: false };
+
+    // Bind state users/modifiers to `this`.
+    this.setEditActivity = this.setEditActivity.bind(this);
+    this.finishEditActivity = this.finishEditActivity.bind(this);
+    this.displayCard = this.displayCard.bind(this);
+  }
+
+  /**
+   * Set the activity into editing mode.
+   */
+  setEditActivity = () => ( this.setState({editing: true}) );
+
+  /**
+   * Set the activity into viewing mode.
+   */
+  finishEditActivity = () => (  this.setState({editing: false}) );
+
+  /**
+   * Display the current activity, either in view or display mode.
+   */
+  displayCard() {
+    let activity = this.props.activity;
+    if (!this.state.editing) { // View mode.
+      return (
+        <Card.Body onClick={this.setEditActivity}>
+          <p>Start time: {time.timestampToFormatted(activity[DB.ACTIVITIES_START_TIME])} </p>
+          <p>End time: {time.timestampToFormatted(activity[DB.ACTIVITIES_END_TIME])} </p>
+        </Card.Body>
+      );
+    } else { // Edit mode.
+      return (
+        // TODO: Save form. (#48)
+        <Form className="activity-editor" onSubmit={this.finishEditActivity}>
+          <Form.Group as={Row} controlId="formActivityTitle">
+            <Col sm={2}><Form.Label>Title:</Form.Label></Col>
+            <Col><Form.Control type="text" placeholder={activity[DB.ACTIVITIES_TITLE]}/></Col>
+          </Form.Group>
+          <Form.Group as={Row} controlId="formActivityStartTime">
+            <Col sm={2}><Form.Label>From:</Form.Label></Col>
+            <Col sm={4}><Form.Control type="date" label="date"/></Col>
+            <Col sm={2}><Form.Control type="time" label="time"/></Col>
+            <Col sm={1}>{timezonePicker()}</Col>
+          </Form.Group>
+          <Form.Group as={Row} controlId="formActivityStartTime">
+            <Col sm={2}><Form.Label>To:</Form.Label></Col>
+            <Col sm={4}><Form.Control type="date" label="date"/></Col>
+            <Col sm={2}><Form.Control type="time" label="time"/></Col>
+            <Col sm={1}>{timezonePicker()}</Col>
+          </Form.Group>
+          <Form.Group as={Row} controlId="formActivityTitle">
```

Comment:
You use formActivityTitle above.
```suggestion
          <Form.Group as={Row} controlId="formActivityDescription">
```

Note: before you commit this change (if you do), check to make sure my email associated with the commit is my @google.

Reply:
Was this changed? It appears unchanged for me.